core/utils/cuckoo_map.h

```diff
@@ -201,11 +200,27 @@ class CuckooMap {
       }
 
       // expand the table as the last resort
-      ExpandBuckets(hasher, eq);
+      ExpandBuckets<std::conditional_t<std::is_move_constructible<V>::value,
+                                       V&&, const V&>>(hasher, eq);
     }
     return entry;
   }
 
+  // Insert/update a key value pair
+  // Return the pointer to the inserted entry
+  template <typename VV>
+  Entry* Insert(const K& key, VV&& value, const H& hasher = H(),
```

Comment:
`template <typename VV> ... VV&& ... std::forward<VV>`: deep C++14 voodoo: https://en.cppreference.com/w/cpp/utility/forward

Discussed offline: maybe should comment that constructor might not be called if the `Insert` fails (returns `nullptr`). We don't depend on this, but if someone does, it could be surprising.